packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -153,12 +174,73 @@ export function substituteFoamVariables(
   return templateText;
 }
 
+function sortTemplatesMetadata(templatesMetadata: object[]) {
+  // Sort by name's existence, name, then path
+  return templatesMetadata.sort((t1, t2) => {
+    if (t1['name'] === undefined && t2['name'] !== undefined) {
+      return 1;
+    }
+
+    if (t1['name'] !== undefined && t2['name'] === undefined) {
+      return -1;
+    }
+
+    if (t1['name'] > t2['name']) {
+      return 1;
+    }
+
+    if (t1['name'] < t2['name']) {
+      return -1;
+    }
+
+    if (t1['templatePath'] > t2['templatePath']) {
+      return 1;
+    }
+
+    if (t1['templatePath'] < t2['templatePath']) {
+      return -1;
+    }
+
+    return 0;
+  });
+}
+
 async function askUserForTemplate() {
   const templates = await getTemplates();
   if (templates.length === 0) {
     return offerToCreateTemplate();
   }
-  return await window.showQuickPick(templates, {
+
+  const templatesMetadata = sortTemplatesMetadata(
```

Comment:
somewhat minor, mostly style: in general in the code we try to provide sorting functions instead of functions that sort (I hope it's clear what I mean).

basically here it would translate into something like:

```
(await Promise.all( .... )).sort(sortMetadataByNameAndPath)
```

thoughts?

Reply:
Makes total sense. Fixed.